cmd/mt-update-ttl/main.go

```diff
@@ -0,0 +1,167 @@
+	}
+
+	session, err := NewCassandraStore(*cassandraAddrs, *cassandraKeyspace, *cassandraConsistency, *cassandraCaPath, *cassandraUsername, *cassandraPassword, *cassandraHostSelectionPolicy, *cassandraTimeout, *cassandraConcurrency, *cassandraRetries, *cqlProtocolVersion, *cassandraSSL, *cassandraAuth, *cassandraHostVerification)
+
+	if err != nil {
+		panic(fmt.Sprintf("Failed to instantiate cassandra: %s", err))
+	}
+
+	update(session, ttl, tableIn, tableOut)
+}
+
+func NewCassandraStore(addrs, keyspace, consistency, CaPath, Username, Password, hostSelectionPolicy string, timeout, concurrency, retries, protoVer int, ssl, auth, hostVerification bool) (*gocql.Session, error) {
+
+	cluster := gocql.NewCluster(strings.Split(addrs, ",")...)
+	if ssl {
+		cluster.SslOpts = &gocql.SslOptions{
+			CaPath:                 CaPath,
+			EnableHostVerification: hostVerification,
+		}
+	}
+	if auth {
+		cluster.Authenticator = gocql.PasswordAuthenticator{
+			Username: Username,
+			Password: Password,
+		}
+	}
+	cluster.Consistency = gocql.ParseConsistency(consistency)
+	cluster.Timeout = time.Duration(timeout) * time.Millisecond
+	cluster.NumConns = concurrency
+	cluster.ProtoVersion = protoVer
+	cluster.Keyspace = keyspace
+	cluster.RetryPolicy = &gocql.SimpleRetryPolicy{NumRetries: retries}
+
+	switch hostSelectionPolicy {
+	case "roundrobin":
+		cluster.PoolConfig.HostSelectionPolicy = gocql.RoundRobinHostPolicy()
+	case "hostpool-simple":
+		cluster.PoolConfig.HostSelectionPolicy = gocql.HostPoolHostPolicy(hostpool.New(nil))
+	case "hostpool-epsilon-greedy":
+		cluster.PoolConfig.HostSelectionPolicy = gocql.HostPoolHostPolicy(
+			hostpool.NewEpsilonGreedy(nil, 0, &hostpool.LinearEpsilonValueCalculator{}),
+		)
+	case "tokenaware,roundrobin":
+		cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(
+			gocql.RoundRobinHostPolicy(),
+		)
+	case "tokenaware,hostpool-simple":
+		cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(
+			gocql.HostPoolHostPolicy(hostpool.New(nil)),
+		)
+	case "tokenaware,hostpool-epsilon-greedy":
+		cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(
+			gocql.HostPoolHostPolicy(
+				hostpool.NewEpsilonGreedy(nil, 0, &hostpool.LinearEpsilonValueCalculator{}),
+			),
+		)
+	default:
+		return nil, fmt.Errorf("unknown HostSelectionPolicy '%q'", hostSelectionPolicy)
+	}
+
+	return cluster.CreateSession()
+}
+
+func getTTL(now, ts, ttl int) int {
+	newTTL := ttl - (now - ts)
+
+	// setting a TTL of 0 or negative, is equivalent to no TTL = keep forever.
+	// we wouldn't want that to happen by accident.
+	// this can happen when setting or shortening a TTL for already very old data.
+	if newTTL < 1 {
+		return 1
+	}
+	return newTTL
+}
+
+func update(session *gocql.Session, ttl int, tableIn, tableOut string) {
+	var key string
+	var ts int
+	var data []byte
+	var query string
+	iter := session.Query(fmt.Sprintf("SELECT key, ts, data FROM %s", tableIn)).Iter()
```

Comment:
does this scale?  If you have a millions of series stored for many months will this cassandra query work, or will it timeout?